app/services/discourse_subscriptions/campaign.rb

```diff
@@ -48,6 +50,27 @@ def create_campaign
 
     protected
 
+    def check_goal_status
+      original_status = SiteSetting.discourse_subscriptions_campaign_goal_met.dup
+      goal = SiteSetting.discourse_subscriptions_campaign_goal
+      goal_type = SiteSetting.discourse_subscriptions_campaign_type
+
+      case goal_type
+      when "Amount"
+        current_volume = SiteSetting.discourse_subscriptions_campaign_amount_raised
+      when "Subscribers"
+        current_volume = SiteSetting.discourse_subscriptions_campaign_subscribers
+      end
+
+      SiteSetting.discourse_subscriptions_campaign_goal_met = current_volume >= goal
```

Comment:
It feels awkward to use site settings for the calculations here. 

Instead, you could maybe use the redis store. We do something similar for storing `last_version_check` and `new_features_last_seen` dates in discourse_updates.rb. 

And I believe you only need to store the `met_date` and clear it if the contributions drop below the 90% threshold.

Reply:
I agree site settings is a bit weird, but since we're already using them to store all the other banner values (per David's suggestion over using a single row table), it made sense to me.

After looking into implementation, my only qualm with using Redis is needing another call to the back-end to grab the data -- which I've tried to avoid. We only have one call to grab contributors for the user avatars, but even then that's an optional setting, so we can't easily piggyback there. Any recommendations on alternatives? 

We could refactor to one general call, I suppose, but it seems excessive for one bit of data.

Reply:
Any thoughts here @pmusaraj?

Reply:
Sorry, I missed your initial reply here. 

I see what you mean about not wanting to make another call to the backend. It makes sense to want to have that info in the main payload. 

I think this can still be done without site settings. One way to do that would be to extend the `SiteSerializer` and add the subscription status there, and then in JS you can access that via `this.site.subscription_status`. We do this in the teams plugin, for example, we pass a list of `group_names` to the frontend. 

(I would also move the date calculation to the backend as well, so in the frontend you would have no logic at all, and that way you only need to pass a boolean show/dontshow variable to the frontend.)

Reply:
@pmusaraj I've fixed this up to use Redis + extending SiteSerializer. Mind another review?